sonar-php-plugin/src/main/java/org/sonar/plugins/php/phpunit/PhpUnitImporter.java

```diff
@@ -19,12 +19,59 @@
  */
 package org.sonar.plugins.php.phpunit;
 
+import com.thoughtworks.xstream.XStreamException;
 import java.io.File;
 import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.sonar.api.batch.sensor.SensorContext;
 
-public interface PhpUnitImporter {
+public abstract class PhpUnitImporter {
 
-  void importReport(File coverageReportFile, SensorContext context, Map<String, Integer> numberOfLinesOfCode);
+  private static final Logger LOGGER = LoggerFactory.getLogger(PhpUnitImporter.class);
+
+  private String reportPathKey;
+  private String msg;
+
+  protected PhpUnitImporter(String reportPathKey, String msg) {
+    this.reportPathKey = reportPathKey;
+    this.msg = msg;
+  }
+
+  public void parseReport(SensorContext context, Map<String, Integer> numberOfLinesOfCode) {
+    String reportPath = context.settings().getString(reportPathKey);
+
+    if (reportPath != null) {
+      File xmlFile = getIOFile(reportPath, context);
+
+      if (xmlFile.exists()) {
+        LOGGER.info("Analyzing PHPUnit " + msg + " report: " + reportPath + " with " + toString());
+        try {
+          importReport(xmlFile, context, numberOfLinesOfCode);
+        } catch (XStreamException e) {
+          throw new IllegalStateException("Report file is invalid, plugin will stop.", e);
+        }
+      } else {
+        LOGGER.info("PHPUnit xml " + msg + " report not found: " + reportPath);
+      }
+    } else {
+      LOGGER.info("No PHPUnit " + msg + " report provided (see '" + reportPathKey + "' property)");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2629)